src/app/shared/components/security-group-builder/security-group-builder.component.ts

```diff
@@ -19,14 +30,25 @@ export class SecurityGroupBuilderComponent implements OnInit {
   @Input() public inputRules: Rules;
   @Output() public onChange = new EventEmitter<Rules>();
 
-  public items: Array<Array<SecurityGroup>>;
+  public securityGroups: Array<Array<SecurityGroup>>;
   public selectedGroupIndex: number;
   public selectedColumnIndex: number;
   public selectedRules: Array<Array<RuleListItem>>;
-  public NetworkRuleTypes = NetworkRuleType;
+
+  public get SecurityGroupBuilderTemplates() {
+    return SecurityGroupBuilderTemplates;
+  }
+
+  public get NetworkRuleType() {
+    return NetworkRuleType;
+  }
+
+  public get NetworkRuleTypes() {
+    return NetworkRuleTypes;
+  }
 
   constructor(private securityGroupService: SecurityGroupService) {
-    this.items = [[], []];
+    this.securityGroups = [[], []];
```

Comment:
Please change it to something like:

this.securityGroups = { available: [], selected: []}
do the same for rules
It will make code easier to read